### ISSUE TITLE
Option to disable jumping into other files

### ISSUE THREAD

**Issue by miliog** [Enter-tainer/typst-preview#149]
I'd like an option to disable jumping between files. In my particular case, I have a template which I do not want to modify or care how the internals look like. So when clicking somewhere on my document inside the preview, it should ideally try to jump to the relevant code within my current file, not dive into my template.

**Comment by Myriad-Dreamin** [comment]
@Enter-tainer you are right, it is a known upstream problem. there will be a stack of source location corresponding to you clicking point in document. However typst doesn't select or synthesize an ideal location from stack, which is hard-to-solve.

**Comment by Myriad-Dreamin** [comment]
We could have some simple option, that rejects to jump to non-main file. But there are space for thinking smarter default setting for complex projects that have multiple files containing main content.
